src/Microsoft.Health.Fhir.Api/Features/Routing/SearchPostReroutingMiddleware.cs

```diff
@@ -42,6 +42,8 @@ public async Task Invoke(HttpContext context)
                         request.Query = new QueryCollection(dic);
                     }
 
+                    request.ContentType = null;
```

Comment:
I'm not quite clear on how this resolves the issue with invalid search parameters. This just looks like it clears out the ContentType and Form if one is sent. Is the form considered a search param in this case?

Reply:
The issue here was that when doing a POST search request with form encoded data the form was being put through a converter by the AspNetCore router after the middleware but before the controller. The converter tried to turn the form keys into JQuery objects, which was triggering the 500. Since we no longer needed the form I removed it here so that the converter wasn't used.

Reply:
Ok, that makes sense. Thanks for the explanation.